### ISSUE TITLE
CI for core branches

### ISSUE THREAD

**Issue by jotego** [jotego/jtcores#285]
**Problem**
When a new core is added to a repository it may break one of the former cores. This occurs because common files are changed to support the new core. These common files may be part of the project or submodules (like JTFRAME itself).

**Proposal**
New core development should be done in a separate branch and only merged into master through a PR in GitHub. GitHub should then run tests on the PR and reject it if they fail

**Tests**
- All cores should compile for MiST, MiSTer and Pocket
- Simulations listed in the `cores/corename/ver/regression.yaml` file should report a **PASS**

**Requirements**
- jotego/jtframe#77 should be fixed first
- jotego/jtcores#286 should be done first
- Per core action files should be generated by the `jtframe update` command (like it's done now)
- Development languages Go and bash scripts
